addons/l10n_no/models/account_invoice.py

```diff
@@ -31,10 +31,9 @@ def calculate_mod10(number):
             for idx, val in enumerate(number[::-1]):
                 for n in str(int(val) * [2, 1][idx % 2]):
                     total += int(n)
-            return number + str((10 - total % 10) % 10)
+            return number + str((10 - total) % 10)
```

Comment:
This cannot be correct

Reply:
12345678 should return 2

Reply:
it returns a correct result.
for 12345678 it returns 2 at the end of the string like 123456782

Reply:
@jva-odoo  Won't total be bigger than 10?

Reply:
Maybe it gives the same result, but I would prefer an easier to understand formula.

Reply:
@jva-odoo What is wrong with str(10 - (total % 10)) ?

Reply:
I know that it can be 0 , which results in 10

Reply:
@jco-odoo I think u need to check this https://github.com/mmcloughlin/luhn

Reply:
noting is wrrong with this str(10 - (total % 10))  or this  str((10 - total) % 10)

Reply:
str(10-0) == "10"  and then you have two numbers instead of 2...  It was ok before, it is just that you need to make clear why it is done that way.

Reply:
ok :+1: